lib/core/src/Cardano/Wallet.hs

```diff
@@ -299,7 +311,7 @@ data WalletLayer s t = WalletLayer
     , listTransactions
         :: DefineTx t
         => WalletId
-        -> ExceptT ErrNoSuchWallet IO [(Tx t, TxMeta)]
+        -> IO [TransactionInfo]
```

Comment:
Hmm... This seems wrong. When there's no wallet, we do expect this layer to return an explicit error, and not an empty list. There's a difference between "this wallet doesn't exist" and "this wallet exists but there's no history" and that's what the error captures :+1:

Reply:
Done: 014ffa52